libcore/src/events/TrainEvent.cpp

```diff
@@ -240,80 +89,89 @@ void TrainEvent::TrainDeparture()
     int room_id, subroom_id;
     std::set<SubRoom *> subRoomsToUpdate;
     // remove temp added walls
-    auto tempWalls = _building->TempAddedWalls[_trainID];
-    for(auto it = tempWalls.begin(); it != tempWalls.end();) {
-        auto wall = *it;
-        if(it != tempWalls.end()) {
-            tempWalls.erase(it);
+    auto tempWalls = _building->GetTrainAddedWalls(_trainID);
+    if(tempWalls.has_value()) {
+        for(auto it = tempWalls.value().begin(); it != tempWalls.value().end();) {
+            auto wall = *it;
+            if(it != tempWalls.value().end()) {
+                tempWalls.value().erase(it);
+            }
```

Comment:
This seems to be quite dangerous, as i am not sure if the iterator is still valid afterwards. Will this be changed in following PRs?

Reply:
It is, will be tackled soon when solving #745